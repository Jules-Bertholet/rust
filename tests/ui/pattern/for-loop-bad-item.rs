struct Qux(i32);

fn good() {
    let mut map = std::collections::HashMap::new();
    map.insert(('a', 'b'), ('c', 'd'));

    for ((_, _), (&mut c, _)) in &mut map {
        if c == 'e' {}
    }
}

fn bad() {
    for Some(Qux(_)) | None in [Some(""), None] {
    //~^ ERROR mismatched types
        todo!();
    }
}

fn main() {}
