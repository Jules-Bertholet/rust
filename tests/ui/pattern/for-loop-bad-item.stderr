error[E0308]: mismatched types
  --> $DIR/for-loop-bad-item.rs:13:14
   |
LL |     for Some(Qux(_)) | None in [Some(""), None] {
   |              ^^^^^^            ---------------- this is an iterator with items of type `Option<&str>`
   |              |
   |              expected `str`, found `Qux`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.
